hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -418,6 +418,12 @@ public class HoodieWriteConfig extends HoodieConfig {
       .sinceVersion("0.9.0")
       .withDocumentation(MarkerType.class);
 
+  public static final ConfigProperty<Boolean> TIMELINE_SERVER_BASED_FLINK_CKP_METADATA = ConfigProperty
+      .key("hoodie.flink.ckp_metadata.timeline_server_based.enable")
+      .defaultValue(true)
```

Comment:
Can we remove the flink keywork here, do we need an option for it, just add a new URI on the timeline server should be fine.

Reply:
Thanks for your review~ 

> Can we remove the flink keywork here
Removing the `flink` keyword is OK. 

> do we need an option for it, just add a new URI on the timeline server should be fine.
I'm mainly following the implementation of timeline-based markers which have an option for enabling timeline-based markers or not, too. 

> you need to handle the diffferent request from various clients in good manner.
The parameter of Ckp metadata Handler is the full ckp metadata path, which already contains the uniqueId of one specific job.  I think it's enough for the multi-client scenario, what do you think?

Reply:
Yeah, should be fine.

Reply:
Ping me again if it is in good shape.

Reply:
@danny0405 It's ready to review now, PTAL

Reply:
`hoodie.ckp_metadata.timeline_server_based.enable` -> `hoodie.instant_state.timeline_server_based.enabled`

Reply:
> hoodie.ckp_metadata.timeline_server_based.enable -> hoodie.instant_state.timeline_server_based.enabled

done